crates/ruff_linter/src/rules/pyupgrade/rules/pep695/use_pep695_type_parameter.rs

```diff
@@ -0,0 +1,258 @@
+use itertools::Itertools;
+
+use ruff_diagnostics::{Applicability, Diagnostic, Edit, Fix, FixAvailability, Violation};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_python_ast::{visitor::Visitor, Expr, ExprSubscript};
+use ruff_python_ast::{StmtClassDef, StmtFunctionDef};
+use ruff_text_size::{Ranged, TextRange};
+
+use crate::checkers::ast::Checker;
+use crate::settings::types::PythonVersion;
+
+use super::{DisplayTypeVars, TypeVar, TypeVarReferenceVisitor};
+
+/// ## What it does
+///
+/// Checks for use of standalone type variables and parameter specifications in generic functions
+/// and classes.
+///
+/// ## Why is this bad?
+///
+/// Special type parameter syntax was introduced in Python 3.12 by [PEP 695] for defining generic
+/// functions and classes. This syntax is easier to read and provides cleaner support for generics.
+///
+/// ## Known problems
+///
+/// [PEP 695] uses inferred variance for type parameters, instead of the `covariant` and
+/// `contravariant` keywords used by `TypeVar` variables. As such, rewriting a `TypeVar` variable to
+/// an in-line type parameter may change its variance.
+///
+/// The rule currently excludes cases where it conceptually should be able to give a diagnostic. In
+/// particular, it skips generic classes with multiple base classes, and it skips generic methods in
+/// generic or non-generic classes.
+///
+/// ## Example
+///
+/// ```python
+/// from typing import TypeVar
+///
+/// T = TypeVar("T")
+///
+///
+/// class GenericClass(Generic[T]): ...
+/// ```
+///
+/// Use instead:
+///
+/// ```python
+/// class GenericClass[T]: ...
```

Comment:
Maybe it's worth making the example slightly more realistic by using the type parameter in the class body?

```suggestion
/// class GenericClass(Generic[T]):
///     var: T
/// ```
///
/// Use instead:
///
/// ```python
/// class GenericClass[T]:
///     var: T
```